paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentSheetCardBrandFilterTest.kt

```diff
@@ -0,0 +1,93 @@
+package com.stripe.android.paymentsheet
+
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.ExperimentalCardBrandFilteringApi
+import com.stripe.android.lpmfoundations.paymentmethod.PaymentSheetCardBrandFilter
+import com.stripe.android.lpmfoundations.paymentmethod.toBrandCategory
+import com.stripe.android.model.CardBrand
+import org.junit.Test
+
+@OptIn(ExperimentalCardBrandFilteringApi::class)
+class PaymentSheetCardBrandFilterTest {
+
+    @Test
+    fun testIsAccepted_allBrandsAccepted() {
+        val filter = PaymentSheetCardBrandFilter(PaymentSheet.CardBrandAcceptance.All)
+
+        for (brand in CardBrand.values()) {
+            assertThat(filter.isAccepted(brand)).isTrue()
+        }
+    }
+
+    @Test
+    fun testIsAccepted_allowedBrands() {
+        val allowedBrands = listOf(
+            PaymentSheet.CardBrandAcceptance.BrandCategory.Visa,
+            PaymentSheet.CardBrandAcceptance.BrandCategory.Mastercard
+        )
+        val filter = PaymentSheetCardBrandFilter(
+            PaymentSheet.CardBrandAcceptance.Allowed(allowedBrands)
+        )
+
+        for (brand in CardBrand.values()) {
+            val brandCategory = brand.toBrandCategory()
+            val isExpectedToBeAccepted =
```

Comment:
I find these sorts of tests to be less useful.

Having logic in the test makes it easy to break, or change without knowing how it actually works.

I typically find specific examples (even if they're not completely exhaustive) to be more valuable.

This is definitely a more preference based comment take it or leave it (feel free to not action this comment).

Reply:
Yeah I feel you. Given the somewhat different handling of null brand categories for allow or disallowed this seemed ~reasonable.